src/Lecture3.hs

```diff
@@ -213,8 +222,23 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr step initial (List1 val []) = step val initial
+  foldr step initial (List1 val (x : xs)) = foldr step (step val initial) (List1 x xs)
```

Comment:
You don't need to implement a manual recursion here, you can reuse the `Foldable` instance for lists 📂